Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2.py

```diff
@@ -0,0 +1,448 @@
+        token = self.authenticate()
+        self.save_jwt_token(token)
+        self.save_access_token(token)
+
+    def save_jwt_token(self, access_token: str):
+        expiration = get_jwt_expiration(access_token)
+        expiration_timestamp = datetime.fromtimestamp(expiration)
+        context = {"access_token": access_token,
+                   "expires_in": expiration_timestamp.strftime(DATE_FORMAT)}
+        set_integration_context(context)
+        demisto.debug(
+            f"New access token that expires in : {expiration_timestamp.strftime(DATE_FORMAT)}"
+            f" was set to integration_context.")
+
+    def authenticate(self):
+        body = {
+            'username': self.username,
+            'password': self.password
+        }
+        new_token = self._http_request(
+            method='POST',
+            url_suffix='/authenticate',
+            json_data=body)
+
+        if not new_token or not new_token.get('access_token'):
+            raise DemistoException(
+                "GuardiCore error: The client credentials are invalid.")
+
+        new_token = new_token.get('access_token')
+        return new_token
+
+    def get_assets(self, url_params: dict):
+        data = self._http_request(
+            method='GET',
+            url_suffix='/assets',
+            params=url_params
+        )
+        return data
+
+    def get_incident(self, url_params: str):
+        data = self._http_request(
+            method='GET',
+            url_suffix=f'/incidents/{url_params}',
+        )
+        return data
+
+    def get_incidents(self, url_params: dict):
+        data = self._http_request(
+            method='GET',
+            url_suffix='/incidents',
+            params=url_params)
+        return data
+
+
+''' HELPER FUNCTIONS '''
+
+
+def get_jwt_expiration(token: str):
+    if "." not in token:
+        return 0
+    jwt_token = base64.b64decode(token.split(".")[1] + '==')
+    jwt_token = json.loads(jwt_token)
+    return jwt_token.get("exp")
+
+
+def calculate_fetch_start_time(last_fetch: Optional[str],
+                               first_fetch: Optional[str]):
+    # Taken from the MicrosoftCloudAppSecurity Integration
+    if last_fetch is None:
+        if not first_fetch:
+            first_fetch = '3 days'
+        first_fetch_dt = parse(first_fetch).replace(tzinfo=utc)  # type:ignore
+        # Changing 10-digits timestamp to 13-digits by padding with zeroes,
+        # since API supports 13-digits
+        first_fetch_time = int(first_fetch_dt.timestamp()) * 1000
+        return first_fetch_time
+    else:
+        return int(last_fetch)
+
+
+def filter_human_readable(results: dict, human_columns=List[str]) -> dict:
+    # Takes in results dict and filters out relevant human_columns.
+    filtered = {}
+    for hc in human_columns:
+        if hc in results:
+            filtered[hc] = results[hc]
+    return filtered
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
```

Comment:
when having fetch_incidents functionality, it is recommended to test it, e.g:
https://github.com/demisto/content/blob/master/Packs/Zimperium/Integrations/Zimperium/Zimperium.py#L215
a better version will be to do that with the fetch params.